docs/pages/docs/core-concepts/caching.mdx

```diff
@@ -152,16 +152,44 @@ determining a cache hit on the `test` task:
 ```
 
 <Callout type="info">
-  Note: `turbo.json` is *always* considered an input. If you modify `turbo.json`, all caches are invalidated.
+  `package.json` is *always* considered an input for tasks in the workspace it lives in.
+  This is because the *definition* of the task itself lives in package.json in the `scripts` key.
+  If you change that, any cached output is considered invalid.
 </Callout>
 
-When you use `turbo` with tools that inline environment variables at build time (e.g. [Next.js](https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser) or [Create React App](https://create-react-app.dev/docs/adding-custom-environment-variables/)), it is important to tell `turbo` about it. Otherwise, you could ship a cached artifact with the wrong environment variables!
+If you want _all_ tasks to depend on certain files, you can declare this dependency in the
+`globalDependencies` array.
 
-Luckily, you can control `turbo`'s cache fingerprinting (also known as "hashing") behavior based on the values of both environment variables and the contents of files:
+```diff
+{
+  "$schema": "https://turborepo.org/schema.json",
++  "globalDependencies": [".env"],
+  "pipeline": {
+    // ...other tasks
+    "test": {
+      "outputs": [], // leave empty to only cache logs
+      "dependsOn": ["build"],
+      "inputs": ["src/**/*.tsx", "src/**/*.ts", "test/**/*.ts"]
+    }
+  }
+}
+```
+
+<Callout type="info">
+  `turbo.json` is *always* considered a global input. If you modify `turbo.json`, all caches
+  are invalidated.
+</Callout>
+
+## Altering Caching Based on Environment Variables
+
+When you use `turbo` with tools that inline environment variables at build time
+(e.g. [Next.js][1] or [Create React App][2]), it is important to tell `turbo` about it.
+Otherwise, you could ship a cached build with the wrong environment variables!
+
+Luckily, you can control `turbo`'s caching behavior (also known as "hashing") based on
```

Comment:
```suggestion
You can control `turbo`'s caching behavior based on
```

@mattpocock just did a pass removing this sort of "how it works" stuff because it doesn't _really_ matter.

Reply:
👍